specifyweb/frontend/js_src/lib/attachmentserverpublic.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+var $ = require('jquery');
+var _ = require('underscore');
+
+var schema         = require('./schema.js');
+var initialContext = require('./initialcontext.js');
+var attachmentserverbase = require('./attachmentserverbase.js');
+
+var settings;
+initialContext.load('attachment_settings.json', data => settings = data);
+
+function functionthatjustpipesthings(notused) {
+    return $.get('/attachment_gw/get_token/', { filename: 'asfdas' });
+}
+
+var attachmentserverpublic = {
+  getThumbnail: function(attachment, scale) {
+      scale || (scale = 256);
+      var style = "max-width:" + scale + "px; " + "max-height:" + scale + "px;";
+
+      var attachmentlocation = attachment.get('attachmentlocation');
+      return functionthatjustpipesthings(attachmentlocation).pipe(function(token) {
+          return $('<img>', {src: `http://localhost:5050/loris/${attachmentlocation}/full/${scale},/0/default.jpg`, style: style});
```

Comment:
If you used a base URL in the settings, then you could use that here:

E.g. `${settings.public_image_server_base_url}/${attachmentlocation}/full/${scale},/0/default.jpg`

Reply:
Good idea!